### ISSUE TITLE
Allow beforeSend callback

### ISSUE THREAD

**Issue by gabceb** [DataDog/browser-sdk#510]
Hi! We are a pretty big Datadog customer looking to adopt DD browser logger. Based on our experience with other FE logger tools it would be great if the sdk provided a callback before the event is sent to the server so we can decide to drop it based on some heuristics (i.e useless browser exceptions we have no way of fixing, errors we want to ignore, other tools to send errors to, etc).

This would be equivalent to the [before send](https://docs.sentry.io/error-reporting/configuration/filtering/#before-send) callback on Sentry.

**Comment by bcaudan** [comment]
Hi @gabceb!

That is an interesting feature indeed, we will work on it soon.

**Comment by dels07** [comment]
any info on this, I got so many XHR request error from algolia instantsearch when user decide to correct their search

**Comment by BenoitZugmeyer** [comment]
We are working on it. We implemented a `beforeSend` callback to edit sensitive data, but doesn't support canceling the data sent yet.

**Comment by FelixGraf** [comment]
@BenoitZugmeyer Is this something one could implement and make a pull request? I would really love to see this feature and could try implementing it (if that's fine for you - at least I'd assume based on the CONTRIBUTION guideline).

```js
datadogRum.init({
    ...,
    beforeSend: (event) => {
        // remove email from view url
        event.view.url = event.view.url.replace(/email=[^&]*/, "email=REDACTED")
    },
    ...
});
```

I think it could be implemented fairly easy by giving the option to return a boolean in the `beforeSend` callback. Depending on the returned value, RUM could either send the event to Datdog or cancel the sending.

**Comment by BenoitZugmeyer** [comment]
We'll have to talk a bit about it internally. The issue is, if we want to filter events, we can't do this at the same place as `beforeSend`, because (for example) when an Error event is generated, it increments the View event `view.error.count` field. If we want to filter-out the Error event, we should make sure the count does not get incremented.

**Comment by BenoitZugmeyer** [comment]
This has been implemented in [`v2.5.5`](https://github.com/DataDog/browser-sdk/blob/master/CHANGELOG.md#v255)

**Comment by gabceb** [comment]
Amazing. Thank you @BenoitZugmeyer
